src/create_disk.sh

```diff
@@ -247,29 +249,49 @@ fi
 # Now that we have the basic disk layout, initialize the basic
 # OSTree layout, load in the ostree commit and deploy it.
 ostree admin init-fs --modern $rootfs
+# Initialize the "stateroot"
+ostree admin os-init "$os_name" --sysroot $rootfs
+
+# Propagate flags into target repository
 if [ "${rootfs_type}" = "ext4verity" ]; then
     ostree config --repo=$rootfs/ostree/repo set ex-fsverity.required 'true'
 fi
-time ostree pull-local --repo $rootfs/ostree/repo "$ostree" "$commit"
-if test -n "${remote_name}"; then
-    deploy_ref="${remote_name}:${ref}"
-    ostree refs --repo $rootfs/ostree/repo --create "${deploy_ref}" "${commit}"
-else
-    deploy_ref=$commit
-fi
-ostree admin os-init "$os_name" --sysroot $rootfs
+
+# Compute kargs
 # Note that $ignition_firstboot is interpreted by grub at boot time,
 # *not* the shell here.  Hence the backslash escape.
 allkargs="$extrakargs \$ignition_firstboot"
-kargsargs=""
-for karg in $allkargs
-do
-	kargsargs+="--karg-append=$karg "
-done
-ostree admin deploy "${deploy_ref}" --sysroot $rootfs --os "$os_name" $kargsargs
 
-deploy_root="$rootfs/ostree/deploy/${os_name}/deploy/${commit}.0"
-test -d "${deploy_root}"
+if test -n "${deploy_container}"; then
+    kargsargs=""
+    for karg in $allkargs
+    do
+    	kargsargs+="--karg=$karg "
+    done
+    rpm-ostree ex-container image deploy --imgref "${deploy_container}" \
+        ${container_imgref:+--target-imgref $container_imgref} \
+        --stateroot "$os_name" --sysroot $rootfs $kargsargs
+    deploy_root=$(ls -d $rootfs/ostree/deploy/${os_name}/deploy/*.0)
```

Comment:
It doesn't match because the deploy code makes a new commit right now.

And yeah this does mean that things like the rpmdb-in-commit-metadata goes away...but I think we should be putting that stuff in the container manifest probably.  This probably wants a highlevel tracker/discussion issue...
Split out to https://github.com/ostreedev/ostree-rs-ext/issues/143